drf_jsonapi/serializers/resources.py

```diff
@@ -432,10 +432,11 @@ def sort(cls, sort_param=None, queryset=None):
         if not sort_param:
             return queryset
 
-        sort_fields = list(filter(None, sort_param.split(',')))
+        sort_fields = list(filter(None, sort_param.replace('.', '__').split(',')))
 
         # validate the sort fields actually exist in the model
-        field_names = [field.name for field in queryset.model._meta.get_fields()]
+        field_names = getattr(cls.Meta, 'sort_fields', cls.Meta.fields)
+        field_names = list(field_names) + ['id']
```

Comment:
This is the same code you touched here: https://github.com/Vacasa/drf-jsonapi/pull/23/files#diff-994122ce09be62e5a94c8569a02a465bR439

Which of these is correct?

Reply:
@chrisbrantley ^

Reply:
The two bits of code essentially do the same thing just in different ways. I thought the version in  #23 was slightly more pythonic.